internal/jimm/jimm.go

```diff
@@ -193,6 +145,92 @@ type OAuthAuthenticator interface {
 	AuthenticateBrowserSession(ctx context.Context, w http.ResponseWriter, req *http.Request) (context.Context, error)
 }
 
+// RoleManager provides a means to manage roles within JIMM.
+type RoleManager interface {
+	// AddRole adds a role to JIMM.
+	AddRole(ctx context.Context, user *openfga.User, roleName string) (*dbmodel.RoleEntry, error)
+	// GetRoleByUUID returns a role based on the provided UUID.
+	GetRoleByUUID(ctx context.Context, user *openfga.User, uuid string) (*dbmodel.RoleEntry, error)
+	// GetRoleByName returns a role based on the provided name.
+	GetRoleByName(ctx context.Context, user *openfga.User, name string) (*dbmodel.RoleEntry, error)
+	// RemoveRole removes the role from JIMM in both the store and authorisation store.
+	RemoveRole(ctx context.Context, user *openfga.User, roleName string) error
+	// RenameRole renames a role in JIMM's DB.
+	RenameRole(ctx context.Context, user *openfga.User, uuid, newName string) error
+	// ListRoles returns a list of roles known to JIMM.
+	// `match` will filter the list fuzzy matching role's name or uuid.
+	ListRoles(ctx context.Context, user *openfga.User, pagination pagination.LimitOffsetPagination, match string) ([]dbmodel.RoleEntry, error)
+	// CountRoles returns the number of roles that exist.
+	CountRoles(ctx context.Context, user *openfga.User) (int, error)
+}
+
+// Option modifies JIMM returned by the constructor.
```

Comment:
Again I think it would be good to explain why this is useful.
```suggestion
// Option modifies the JIMM returned by the constructor.
// This is useful to create mock domain objects, e.g. replace the real RoleManager with a mock 
// object in tests.
```